dao/dao.go

```diff
@@ -67,22 +70,213 @@ func New(s server, a auth, opts *opts) (*DAO, error) {
 }
 
 // Save stores data into table in clickhouse database
-func (p *DAO) Save(data string) error {
-	if data == "" {
-		return fmt.Errorf("clickhouse invalid data for table %s: %s", p.table, data)
-	}
+// nolint: funlen
+func (p *DAO) Save(theEvent beacon.RumEvent) error {
 	query := fmt.Sprintf(
-		"INSERT INTO %s SETTINGS input_format_skip_unknown_fields = true FORMAT JSONEachRow %s",
+		`INSERT INTO %s (
+			created_at,
+			hostname,
+			event_type,
+			browser_name,
+			browser_version,
+			ua_vnd,
+			ua_plt,
+			device_type,
+			device_manufacturer,
+			operating_system,
+			operating_system_version,
+			user_agent,
+			next_hop_protocol,
+			visibility_state,
+			session_id,
+			session_length,
+			url,
+			connect_duration,
+			dns_duration,
+			first_byte_duration,
+			redirect_duration,
+			redirects_count,
+			first_contentful_paint,
+			first_paint,
+			cumulative_layout_shift,
+			first_input_delay,
+			largest_contentful_paint,
+			geo_country_code,
+			geo_city_name,
+			page_id,
+			data_saver_on,
+			boomerang_version,
+			screen_width,
+			screen_height,
+			dom_res,
+			dom_doms,
+			mem_total,
+			mem_limit,
+			mem_used,
+			mem_lsln,
+			mem_ssln,
+			mem_lssz,
+			scr_bpp,
+			scr_orn,
+			cpu_cnc,
+			dom_ln,
+			dom_sz,
+			dom_ck,
+			dom_img,
+			dom_img_uniq,
+			dom_script,
+			dom_iframe,
+			dom_link,
+			dom_link_css,
+			mob_etype,
+			mob_dl,
+			mob_rtt
+		) VALUES(
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?
+		)`,
 		p.table,
-		data,
 	)
-	err := p.conn.AsyncInsert(context.Background(), query, false)
```

Comment:
@ivan-stankov-golang is there a way to keep AsyncInsert? We need it for performance reasons. AsyncInsert tells to ClickHouse to batch inserts internally and flush them when appropriate.

Reply:
@ivan-stankov-golang it seems that the new float field we are adding is called `mob_rtt` this field was added as float by mistake. We could convert it to Nullable(UInt16).